daemon/daemon_unix.go

```diff
@@ -1211,28 +1211,18 @@ func setupRemappedRoot(config *config.Config) (*idtools.IdentityMapping, error)
 		// update remapped root setting now that we have resolved them to actual names
 		config.RemappedRoot = fmt.Sprintf("%s:%s", username, groupname)
 
-		// try with username:groupname, uid:groupname, username:gid, uid:gid,
+		// try with username and uid,
 		// but keep the original error message (err)
-		mappings, err := idtools.NewIdentityMapping(username, groupname)
+		mappings, err := idtools.NewIdentityMapping(username)
 		if err == nil {
 			return mappings, nil
```

Comment:
always lookup by both name and by number, and concat them

Reply:
There is no need of concating right? Because there will be only one line corresponding to a username or uid. So if search by username gives an error, (which means no entry in the subordinate file), then it will be using either uid or no mapping will be present. 
There cannot be a case where both username and uid are used together right?

Reply:
there can be multiple lines

Reply:
suppose user1 (uid : 1001)

Can there be an entry like
```
user1:100000:65536
1001:200000:65536
```

Reply:
nevermind, it can happen. will do the concatenation

Reply:
In that case, I think the first method you suggested will be better.
i.e `NewIdentityMapping(usrename)` will be called only with a username. The method is guaranteed  to return all mappings, irrespective of whether the system uses a uid, or usrename, or both.

Reply:
Have made the requested changes. PTAL.